### ISSUE TITLE
[Wayland, libdecor] Wrong size of maximized window with hidden titlebar

### ISSUE THREAD

**Issue by polter-rnd** [fltk/fltk#474]
GTK3 allows CSS customizations of headerbar as well. It could be hidden, e.g. for maximized windows (`~/.config/gtk-3.0/gtk.css`):

```
.maximized .titlebar.default-decoration {
  margin: -200px 0 0;
  opacity: 0;
}
```

This approach is used by different GNOME-shell extensions, such as [unite](https://github.com/hardpixel/unite-shell) or [pixel-saver](https://github.com/pixel-saver/pixel-saver) to hide decorations for maximized windows.

Issue is that with this CSS setting, maximized FLTK window on Wayland will have gap at the bottom:
![Screenshot from 2022-07-28 17-21-38](https://user-images.githubusercontent.com/6562863/181545947-aeaf7e4c-0bc4-474e-9fe2-16001fa846fb.png)

On X11 it is handled correctly:
![Screenshot from 2022-07-28 17-23-02](https://user-images.githubusercontent.com/6562863/181543889-a6a0e5df-e4ef-436f-ab99-d606ee8194ac.png)

How to reproduce: put the CSS lines above to `~/.config/gtk-3.0/gtk.css`, run any FLTK example with Wayland+libdecor support and try to maximize it.

**Comment by ManoloFLTK** [comment]
How can the window be un-maximized with this setting ?

**Comment by polter-rnd** [comment]
Like this:

https://user-images.githubusercontent.com/6562863/181624078-7b571672-cdf9-4327-ad7a-64770d195597.mp4

**Comment by ManoloFLTK** [comment]
OK. I see now. Thanks.

Do you have any hint about an API that would allow to check whether a maximized window's titlebar should be shown or hidden?  Or, in other words, whether file `gtk.css` has been customized for maximized windows to hide titlebars ?

The context is that such titlebars are created by this GTK function :  `gtk_header_bar_new()`

This issue belongs to libdecor, really, not to FLTK which merely uses libdecor. More specifically it belongs to the GTK plugin of libdecor which is, as of today, a PR to libdecor. Please, feel free to submit an issue to the source git repo for the [GTK libdecor plugin](https://gitlab.gnome.org/christian-rauch/libdecor/-/tree/gtk_cairo_single) [The GTK plugin is in branch `gtk_cairo_single` of that repo].

**Comment by polter-rnd** [comment]
It happens because GTK titlebar height is being being calculated before the new window state is applied, so it could be outdated in some cases. E.g. in case of custom CSS, they are being applied after the height is calculated, so the content size is wrong as well.

I've created a merge request upstream: [christian-rauch/libdecor!10](https://gitlab.gnome.org/christian-rauch/libdecor/-/merge_requests/10)

**Comment by ManoloFLTK** [comment]
This is now fixed in FLTK (28981f6) following import of the fix in libdecor. Many thanks to @polter-rnd .
